tests/tests/test_decommission.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2021 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import pytest
+
+from ..common_setup import standard_setup_one_client
+from ..MenderAPI import authentication, devauth
+from .common_connect import wait_for_connect
+from .mendertesting import MenderTesting
+
+
+class TestDecommission(MenderTesting):
+    """Tests the decommission and re authentication functionality"""
+
+    def test_decommission(self, standard_setup_one_client):
+        mender_device = standard_setup_one_client.device
+
+        # accept the device
+        devauth.accept_devices(1)
+
+        # Check from client side that the authorization was successful
+        out = mender_device.run(
+            "journalctl -u %s -n 6" % mender_device.get_client_service_name()
```

Comment:
Will 6 lines be enough? It seem to be a bit weak, and we might end up with an unstable test that passes some times and fails others. Can we think about a more robust way to find this?

Reply:
This check is the most important of the test and it is copy-pasted several times; I suggest we refactor it to a helper method in this class.

Reply:
> Will 6 lines be enough? It seem to be a bit weak, and we might end up with an unstable test that passes some times and fails others. Can we think about a more robust way to find this?

I changed it to check the entire journal, but it deletes the portion of the journal that has already been used earlier in the test. It is probably more robust.